core/src/main/java/org/mule/runtime/core/privileged/processor/chain/AbstractMessageProcessorChain.java

```diff
@@ -123,7 +125,23 @@ public CoreEvent process(CoreEvent event) throws MuleException {
     for (Processor processor : getProcessorsToExecute()) {
       // Perform assembly for processor chain by transforming the existing publisher with a publisher function for each processor
       // along with the interceptors that decorate it.
-      stream = stream.transform(applyInterceptors(interceptors, processor));
+      stream = stream.transform(applyInterceptors(interceptors, processor))
+          // #1 Register local error hook to wrap exceptions in a MessagingException maintaining failed event.
+          .subscriberContext(context -> context.put("reactor.onOperatorError.local",
+                                                    resolveMessagingExceptionFunction(processor)))
+          // #2 Register continue error strategy to handle errors without stopping the stream.
+          .errorStrategyContinue((throwable, event) -> {
```

Comment:
this is somewhat similar to `resolveMessagingExceptionFunction`. Can the common functionality be extracted?